django/contrib/staticfiles/storage.py

```diff
@@ -451,9 +453,18 @@ def stored_name(self, name):
         raise ValueError("The name '%s' could not be hashed with %r." % (name, self))
 
 
+_staticfiles_json_entropy = (
+    base64.b32encode(  # for good readability and zero trouble with URLs
+        salted_hmac(
+            key_salt="dummy", value="dummy", secret=None  # i.e. use settings.SECRET_KEY
```

Comment:
> the salt is for namespacing to prevent reuse, so set a proper value instead of dummy.

Changing to `staticfiles_json` with upcoming push…

> Also accessing settings on the module level (or generally "code execution") is a bad idea. Move that into a function etc.

I don't think there's a way around that when `HashedFilesMixin.manifest_name` is effectively filled at module import time. Any ideas how?

Reply:
Make it a property? That said don't waste to much time on the code for now, this will most likely require some further design discussion first.

Reply:
> Make it a property?

I was considering a property but noticed that we're currently class-based while a property would be instance-based, so e.g. direct access to `HashedFilesMixin.manifest_name` would no longer yield a string. That would break at least some backwards compatibility and hence existing code, potentially.

> That said don't waste to much time on the code for now, this will most likely require some further design discussion first.

That's fair.